### PR TITLE
Refactor Fingerprint classes: Javadoc and annotations

### PR DESCRIPTION
**Pull request by oleg-nenashev**
During the usage of Fingerprint classes I've got into several undocumented cases. This PR improves the code's self-documentation

@reviewbybees

### CODE REVIEW

File: core/src/main/java/hudson/model/Fingerprint.java [hunk 1]
```diff
@@ -1095,8 +1111,11 @@ public int compare(FingerprintFacet o1, FingerprintFacet o2) {
 
     /**
      * Finds a facet of the specific type (including subtypes.)
+     * @param <T> Class of the {@link FingerprintFacet}
+     * @return First matching facet of the specified class
```
Comment (KostyaSha) [inline core/src/main/java/hudson/model/Fingerprint.java]: First by time?

Comment (jglick) [inline core/src/main/java/hudson/model/Fingerprint.java]: Seems to be "some" not "first", which is fine when you are only assuming one instance per class.

**Comment by jglick** [comment]
:+1:

File: core/src/main/java/hudson/model/Fingerprint.java (hunk 1 above)
Comment (KostyaSha) [inline core/src/main/java/hudson/model/Fingerprint.java]: Right. First changed lines were about time, so i thought it some kind related.

**Comment by KostyaSha** [comment]
:+1:

**Pull request merged**